cfme/tests/infrastructure/test_snapshot.py

```diff
@@ -109,33 +109,41 @@ def test_verify_revert_snapshot(test_vm, provider, soft_assert, register_event,
         'password': credentials[provider.data['full_template']['creds']]['password'],
         'hostname': ip
     }
-    with SSHClient(**ssh_kwargs) as ssh_client:
-        ssh_client.run_command('touch snapshot1.txt')
-        snapshot1.create()
-        ssh_client.run_command('touch snapshot2.txt')
-        snapshot2 = new_snapshot(test_vm)
-        snapshot2.create()
-        snapshot1.revert_to()
+    ssh_client = SSHClient(**ssh_kwargs)
```

Comment:
I can't use `with` statement here. `SSHClient` has in its `__enter__` method `connect`, which is failing. `wait_for` is suited for masking these failures and thus I can wait for successful ssh connection.
In this case, ssh connection is failing because the VM we just created on a provider does not yet have its IP assigned, although it's already visible in CFME UI.